src/app/component/ProductCompare/ProductCompare.container.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+import { ProductItemsType } from 'Type/ProductList';
+
+import ProductCompare from './ProductCompare.component';
+
+export const ProductCompareDispatcher = import(
+    /* webpackMode: "lazy", webpackChunkName: "dispatchers" */
+    'Store/ProductCompare/ProductCompare.dispatcher'
+);
+
+/** @namespace Component/ProductCompare/Container/mapStateToProps  */
+export const mapStateToProps = (state) => ({
+    products: state.ProductCompareReducer.products,
+    isLoading: state.ProductCompareReducer.isLoading,
+    device: state.ConfigReducer.device
+});
+
+/** @namespace Component/ProductCompare/Container/mapDispatchToProps  */
+export const mapDispatchToProps = (dispatch) => ({
+    fetchCompareList: () => ProductCompareDispatcher.then(
+        ({ default: dispatcher }) => dispatcher.getCompareList(dispatch)
+    ),
+    clearCompareList: () => ProductCompareDispatcher.then(
+        ({ default: dispatcher }) => dispatcher.clearComparedProducts(dispatch)
+    )
+});
+
+/** @namespace Component/ProductCompare/Container */
+export class ProductCompareContainer extends PureComponent {
+    static propTypes = {
+        fetchCompareList: PropTypes.func.isRequired,
+        clearCompareList: PropTypes.func.isRequired,
+        isLoading: PropTypes.bool,
+        products: ProductItemsType
+    };
+
+    static defaultProps = {
+        isLoading: false,
+        products: []
+    };
+
+    containerFunctions = {
+        getAttributes: this.getAttributes.bind(this),
+        clearCompareList: this.clearCompareList.bind(this)
+    };
+
+    componentWillMount() {
```

Comment:
This is unsafe method, no? https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

Reply:
Fixed